### PR TITLE
[Typing][C-78] Add type annotations for `python/paddle/incubate/nn/functional/masked_multihead_attention.py`

### PR DESCRIPTION
**Pull request by haoyu2022**
### PR Category

User Experience

### PR Types

Improvements

### Description

类型标注：
- python/paddle/incubate/nn/functional/masked_multihead_attention.py

### Related links
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

File: python/paddle/incubate/nn/functional/masked_multihead_attention.py [hunk 1]
```diff
@@ -12,9 +12,64 @@
...
+    out_smooth: Tensor | None = ...,
+    seq_len: int = ...,
+    rotary_emb_dims: int = ...,
+    use_neox_rotary_style: bool = ...,
+    compute_dtype: str = ...,
+    out_scale: float = ...,
+    quant_round_type: int = ...,
+    quant_max_bound: float = ...,
+    quant_min_bound: float = ...,
+) -> tuple[Tensor, Tensor]: ...
```
Comment (megemini) [inline python/paddle/incubate/nn/functional/masked_multihead_attention.py]:
```suggestion
) -> tuple[Tensor, Tensor, Tensor]: ...
```

File: python/paddle/incubate/nn/functional/masked_multihead_attention.py [hunk 2]
```diff
@@ -12,9 +12,64 @@
...
+    out_smooth: Tensor | None = ...,
+    seq_len: int = ...,
+    rotary_emb_dims: int = ...,
+    use_neox_rotary_style: bool = ...,
+    compute_dtype: str = ...,
+    out_scale: float = ...,
+    quant_round_type: int = ...,
+    quant_max_bound: float = ...,
+    quant_min_bound: float = ...,
+) -> Tensor: ...
```
Comment (megemini) [inline python/paddle/incubate/nn/functional/masked_multihead_attention.py]:
```suggestion
) -> tuple[Tensor, Tensor]: ...
```

**Review by megemini — COMMENTED** [review]
文档中的参数类型不对，也改一下吧：

``` python
cache_kvs (list(Tensor)|tuple(Tensor)): The cache structure tensors ...
```

改为 

``` python
cache_kv (Tensor): The cache structure tensors ...
```

File: python/paddle/incubate/nn/functional/masked_multihead_attention.py [hunk 3]
```diff
@@ -44,7 +99,7 @@ def masked_multihead_attention(
 
     Args:
         x (Tensor): The input tensor could be 2-D tensor. Its shape is [batch_size, 3 * num_head * head_dim].
-        cache_kvs (list(Tensor)|tuple(Tensor)): The cache structure tensors for the generation model. Its shape is [2, batch_size, num_head, max_seq_len, head_dim].
+        cache_kvs (Tensor): The cache structure tensors for the generation model. Its shape is [2, batch_size, num_head, max_seq_len, head_dim].
```
Comment (megemini) [inline python/paddle/incubate/nn/functional/masked_multihead_attention.py]:
```suggestion
        cache_kv (Tensor): The cache structure tensors for the generation model. Its shape is [2, batch_size, num_head, max_seq_len, head_dim].
```

Comment (haoyu2022) [reply python/paddle/incubate/nn/functional/masked_multihead_attention.py]: 没注意到后面的“s”，多谢指正。😊

**Review by megemini — APPROVED** [review]
LGTM ～

**Review by luotao1 — APPROVED** [review]

**Pull request merged**